gobblin-utility/src/main/java/gobblin/util/SerializationUtils.java

```diff
@@ -102,4 +111,52 @@
       closer.close();
     }
   }
+
+  /**
+   * Serialize a {@link State} instance to a file.
+   *
+   * @param fs the {@link FileSystem} instance for creating the file
+   * @param jobStateFilePath the path to the file
+   * @param state the {@link State} to serialize
+   * @param <T> the {@link State} object type
+   * @throws IOException if it fails to serialize the {@link State} instance
+   */
+  public static <T extends State> void serializeState(FileSystem fs, Path jobStateFilePath, T state)
+      throws IOException {
+    Closer closer = Closer.create();
+
+    try {
+      OutputStream os = closer.register(fs.create(jobStateFilePath));
+      DataOutputStream dataOutputStream = closer.register(new DataOutputStream(os));
+      state.write(dataOutputStream);
+    } catch (Throwable t) {
+      throw closer.rethrow(t);
+    } finally {
+      closer.close();
+    }
+  }
+
+  /**
+   * Deserialize/read a {@link State} instance from a file.
```

Comment:
Please clarify if the state passed to `deserializeState` contains properties, whether those properties will be removed.